### ISSUE TITLE
Linked places should not appear in place_addressline

### ISSUE THREAD

**Issue by pearsh** [osm-search/Nominatim#483]
I did some reverse geocoding task using points located in Japan. The result seems not right.
For example [http://nominatim.openstreetmap.org/reverse?format=xml&lat=32.687&lon=129.847&zoom=18&addressdetails=1](url)  shows this point in  Nishi Sonogi County. This is not right.  Nishi Sonogi County should be the same level as Nagasaki city and it does not cover Nagasaki.

**Comment by pearsh** [comment]
Sorry I don't know how to insert correct url of Nominatim service. So I paste the result below:
<city>Nagasaki</city>
<county>Nishi Sonogi County</county>
<region>Nagasaki Prefecture</region>
City Nagasaki and Region Nagasaki Prefecture is right.  Nishi Sonogi County does not cover given latlng. Seems Nominatim treat county as a higher level than city in Japan.

**Comment by lonvia** [comment]
The place node for 'Nishi Sonogi County' is correctly linked to the corresponding boundary relation and should therefore not appear in the address listing, still it is there. I can reproduce a similar issue in my freshly imported test DB, so it's not an issue with updates.

**Comment by lonvia** [comment]
Non-technical summary: I've pushed a hacky fix that hides the bad county for now. Still need to implement a proper fix that removes the county completely, so the issue remains open.

So what happens is that the place node was already indexed and therefore appears in `location_area_large`. When it later linked to the boundary, it is not removed from `location_area_large`. The address search in `getNearFeatures()` does not check for the linked_place_id flag either and so it ends up in `place_addressline` as a valid address part.

I've pushed a quick and dirty fix onto the osm.org which simply hides linked nodes. That will fix the originally reported issue but isn't really a final solution. Firstly, the county still appears in the search terms and secondly it may cause valid address parts to be hidden.

Proper fix: either make sure that linked places get removed from `location_area_large` (preferred) or change `getNearFeatures()` to ignore linked places. In any case, it means that a change of linked status requires all places that contain the linkee to be updated.

I'll change the title to reflect the technical issue.
